src/interpreter/InterpretAdditionalProperties.ts

```diff
@@ -0,0 +1,19 @@
+import { CommonModel } from 'models';
+import { Schema } from 'models/Schema';
+import { Interpreter } from './Interpreter';
+import { isModelObject } from './Utils';
+
+/**
+ * Interpreter function for JSON Schema draft 7 additionalProperties keyword.
+ * 
+ * @param schema
+ * @param model
+ * @param interpreter
+ */
+export default function interpretAdditionalProperties(schema: Schema | boolean, model: CommonModel, interpreter : Interpreter) {
+  if (typeof schema === 'boolean' || !isModelObject(model)) return;
+  const newModels = interpreter.interpret(schema.additionalProperties || true);
```

Comment:
If you have `true` then you never go to `15` line so

```ts
const newModels = interpreter.interpret(schema.additionalProperties);
```

is enough.

Reply:
There is a difference between the schema being boolean and the additionalProperty schema being a boolean.

Not sure what you are suggesting here? 🤔

Reply:
waiiit, `schema` should not have type `boolean | schema`... 

Gonna change it